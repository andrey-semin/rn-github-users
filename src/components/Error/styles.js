import { StyleSheet } from 'react-native'

export default StyleSheet.create({
  root: {
    justifyContent: 'center',
    alignItems: 'center'
  }
})
