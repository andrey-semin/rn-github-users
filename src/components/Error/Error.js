import React from 'react'
import PropTypes from 'prop-types'
import { View, Text } from 'react-native'

import styles from './styles'

const Error = ({ label }) => (
  <View style={styles.root}>
    <Text>{label}</Text>
  </View>
)

Error.propTypes = {
  label: PropTypes.string.isRequired
}

export default Error
