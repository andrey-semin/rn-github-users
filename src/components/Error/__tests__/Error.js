import React from 'react'
import renderer from 'react-test-renderer'

import Error from '../Error'

it('Error renders correctly', () => {
  const tree = renderer.create(<Error label="some error text" />).toJSON()
  expect(tree).toMatchSnapshot()
})
