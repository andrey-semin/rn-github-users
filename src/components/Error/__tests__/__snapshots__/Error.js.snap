// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    some error text
  </Text>
</View>
`;
